lib/Doctrine/ORM/Query/Parser.php

```diff
@@ -2541,18 +2354,8 @@ public function ConditionalPrimary()
      *      InExpression | NullComparisonExpression | ExistsExpression |
      *      EmptyCollectionComparisonExpression | CollectionMemberExpression |
      *      InstanceOfExpression
-     *
-     * @return AST\BetweenExpression|
-     *         AST\CollectionMemberExpression|
-     *         AST\ComparisonExpression|
-     *         AST\EmptyCollectionComparisonExpression|
-     *         AST\ExistsExpression|
-     *         AST\InExpression|
-     *         AST\InstanceOfExpression|
-     *         AST\LikeExpression|
-     *         AST\NullComparisonExpression
-     */
-    public function SimpleConditionalExpression()
+     */
+    public function SimpleConditionalExpression(): AST\ExistsExpression|AST\BetweenExpression|AST\LikeExpression|AST\InListExpression|AST\InSubselectExpression|AST\InstanceOfExpression|AST\CollectionMemberExpression|AST\NullComparisonExpression|AST\EmptyCollectionComparisonExpression|AST\ComparisonExpression
```

Comment:
Should I simplify this one to `AST\Node`?

Reply:
Depends on how we answer my question regarding `final`. 🙂